code/game/objects/items/weapons/twohanded.dm

```diff
@@ -405,6 +406,32 @@
 	if(wielded)
 		. = ..()
 
+
+/obj/item/weapon/twohanded/required/chainsaw/doomslayer
+	name = "OOOH BABY"
+	desc = "<span class='warn'>VRRRRRRR!!!</span>"
+	force = 20
+	armour_penetration = 100
+
+/obj/item/weapon/twohanded/required/chainsaw/doomslayer/attack_self(mob/user)
+	on = !on
+	user << "You viciously yank the cord dangling from \the [src], [on ? "and it roars to life, screaming along with your bloodlust." : "grows dormant."]"
+	force = on ? 80 : 20
```

Comment:
I thought it was a bit spaghetti. that'd require a bit of a rewrite of the chainsaw tho because that var is only used for energy weapons. i might just replace it with a reskinned energy axe.